immuneML/reports/encoding_reports/DesignMatrixExporter.py

```diff
@@ -50,8 +61,35 @@ def _generate(self) -> ReportResult:
         return ReportResult(self.name, output_tables=[matrix_result], output_text=[details_result, label_result])
 
     def _export_matrix(self) -> ReportOutput:
+        """Create a file for the design matrix in the desired format."""
+        
         data = self._get_data()
-        file_path = self._save_to_file(data, self.result_path / "design_matrix")
+        file_path = self.result_path / "design_matrix"
+
+        # Check that the requested extension is supported.
+        assert self.format.endswith(("hdf5", "hdf5.zip", "csv", "csv.zip", "npy", "npy.zip")), \
+            f'Output format {self.format} not recognised for the Encoding Report'
+
+        # Use h5py to create a hdf5 file.
+        if self.format.endswith(("hdf5", "hdf5.zip")):
+            file_path = file_path.with_suffix(".hdf5")
+            with h5py.File(file_path, 'w') as hf_object:
+                hf_object.create_dataset(str(file_path), data=data)
+        # Use numpy to create a csv ord npy file.
+        elif self.format.endswith(("csv", "csv.zip")):
```

Comment:
In some cases, it might not be possible to store the data as a csv file [e.g., when used with OneHotEncoder] because it has more than 2 dimensions [e.g., one-hot encoding sequences in a repertoire across a repertoire dataset]. That's why the previous function _save_to_file() checked for the shape of data. Could include that in the new version as well? :) Thank you! 🙏

Reply:
done :)